jsign-core/src/main/java/net/jsign/SignerHelper.java

```diff
@@ -496,6 +499,32 @@ private void attach(Signable signable, File detachedSignature) throws IOExceptio
         // todo warn if the hashes don't match
     }
 
+    private void printSignatures(Signable signable) throws IOException {
+        List<CMSSignedData> signatures = signable.getSignatures();
+        for (int i = 0; i < signatures.size(); i++) {
+            CMSSignedData signature = signatures.get(i);
+            SignerInformation signerInformation = signature.getSignerInfos().iterator().next();
+
+            String digestAlgorithmName = new DefaultAlgorithmNameFinder().getAlgorithmName(signerInformation.getDigestAlgorithmID());
+            SignerId signerId = signerInformation.getSID();
+            X509CertificateHolder cert = (X509CertificateHolder) signature.getCertificates().getMatches(signerId).iterator().next();
+
+            byte[] digest = signable.computeDigest(DigestAlgorithm.of(signerInformation.getDigestAlgorithmID().getAlgorithm()));

```

Comment:
This is the hash of the file, I think it should be the hash of the message signed